_posts/2021-09-23-reassign-partitions.md

```diff
@@ -0,0 +1,282 @@
+The answer to this is running commands from within a broker is not a good practice and is insecure. So it is always better to avoid running the command from a broker pod.
+
+So now it's time to get our interactive pod up and running. You can use the following command:
+
+```sh
+kubectl run --restart=Never --image=quay.io/strimzi/kafka:latest-kafka-2.8.0 <interactive_pod_name> -- /bin/sh -c "sleep 3600"
+```
+
+Wait till the pod gets into the `Ready` state. Once the pod gets into `Ready` state, now our next step will be to copy the CA certificate to the interactive pod container:
+
+```sh
+kubectl cp ca.p12 <interactive_pod_name>:/tmp
+```
+
+We can now proceed to fetch the CA certificate and password for the user. User CA certificate and password are extracted from the User Secret like this :
+
+```sh
+kubectl get secret <kafka_user> -o jsonpath='{.data.user\.p12}' | base64 -d > user.p12
+```
+
+```sh
+kubectl get secret <kafka_user> -o jsonpath='{.data.user\.password}' | base64 -d > user.password
+```
+
+Copy the User CA certificates to the interactive pod container.
+
+```sh
+kubectl cp user.p12 <interactive_pod_name>:/tmp
+```
+
+The interactive pod now has both user and cluster CA certificates. 
+
+Next, we will configure the Kafka clients to connect to our Kafka cluster and authenticate.
+The truststore and the keystore location are mapped to their respective location (which is the folder inside the interactive pod container, since all of our commands will be executed inside the shell process of the interactive pod).
+And the configuration provider will use these values.
+We will also configure the `bootstrap.servers` option to make it easier to change address to connect to the Kafka cluster.
+
+```properties
+bootstrap.servers=<kafka_cluster_name>-kafka-bootstrap:9093 
+security.protocol=SSL 
+ssl.truststore.location=/tmp/ca.p12 
+ssl.truststore.password=<truststore_password_present_in_ca_password_file> 
+ssl.keystore.location=/tmp/user.p12 
+ssl.keystore.password=<keystore_password_present_in_user_password_file> 
+```
+
+Now we will copy the `config.properties` file to the interactive pod container:
+
+```sh
+kubectl cp config.properties <interactive_pod_name>:/tmp/config.properties
+```
+
+Lets create a `topics.json` file now. As we discussed above this file will have the topics that we need to reassign:
+
+```json
+{
+  "version": 1,
+  "topics": [
+    { "topic": "my-topic-1"},
+    { "topic": "my-topic-2"}
+  ]
+}
+```
+
+After the creation of this file, copy it to the interactive pod container since we will be running all of our commands from there:
+
+```sh
+kubectl cp topics.json <interactive_pod_name>:/tmp/topics.json
+```
+
+We can now start a shell process in our interactive pod container and run the command inside it to generate our proposal `reassignment.json` data.
+Let us start the shell process:
+
+```sh
+kubectl exec -n <namespace> -ti <interactive_pod_name> /bin/bash
```

Comment:
You did not use namespace in the `kubectl run` and `kubectl cp` commands. So maybe you should avoid it here as well?